scipy/stats/_stats_py.py

```diff
@@ -4372,38 +4372,45 @@ def alexandergovern(*samples, nan_policy='propagate'):
     the alternative.
 
     """
-    samples = _alexandergovern_input_validation(samples, nan_policy)
-
-    if np.any([(sample == sample[0]).all() for sample in samples]):
-        msg = "An input array is constant; the statistic is not defined."
-        warnings.warn(stats.ConstantInputWarning(msg), stacklevel=2)
-        return AlexanderGovernResult(np.nan, np.nan)
+    samples = _alexandergovern_input_validation(samples, nan_policy, axis)
+    samples = [np.moveaxis(sample, axis, -1) for sample in samples]
```

Comment:
Important for the rest of the calculation: 
- `axis` 0 corresponds with separate samples
- `axis` -1 is along a given sample
- any other axes are batch dimensions.